src/ec/packages/ec-component-site-header-core/site-header-core.test.js

```diff
@@ -44,6 +44,15 @@ describe('EC - Site Header Core', () => {
 
       return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
     });
+
+    test('with missing input data and degug enabled returns the right warning message', () => {
+      expect.assertions(1);
+
+      const dataCompliance = { ...englishData, _compliance_: true };
+      dataCompliance.icon_file_path = '';
```

Comment:
No no, keep doing it, or i will check myself, as i wrote you in another pr the intention at the beginning (wrong) was more to test the functionality than the different use cases ;)

Reply:
I don't think having a single test case instead of 10 (for example) is wrong. It's simply not covering the code as well and when you modify an if statement something can be missed because it's not tested.

https://citnet.tech.ec.europa.eu/CITnet/jira/browse/FRONT-1167 is created for investigating and potentially refactor tests. I feel the main reason there aren't more tests at the moment is that the surface (the snapshots) are too big and hard to work with.